### ISSUE TITLE
Support for Azure Firewall TLS Settings

### ISSUE THREAD

**Issue by alexbakervr** [hashicorp/terraform-provider-azurerm#12140]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please can we have a setting within azurerm_firewall  or azurerm_firewall_policy which allows us to enable TLS inspection as I cannot see an option for this anywhere within the Terraform documentation. 

TLS inspection is still in preview from Microsoft but it would be helpful to have the setting within Terraform as it is already available within the Azure portal.

### New or Affected Resource(s)

* azurerm_firewall OR azurerm_firewall_policy

### Potential Terraform Configuration

```hcl
resource "azurerm_firewall" "example" {
  name                              = "testfirewall"
  location                          = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  tls_enabled                    = "true"

  ip_configuration {
    name                 = "configuration"
    subnet_id            = azurerm_subnet.example.id
    public_ip_address_id = azurerm_public_ip.example.id
  }
}
```
OR 
```hcl
resource "azurerm_firewall_policy" "example" {
  name                = "example"
  resource_group_name = "example"
  location            = "West Europe"
  tls_enabled      = "true"
  
}
```
### References

* #0000

**Comment by pckls** [comment]
I believe this has been fixed in #12769

**Comment by mybayern1974** [comment]
@alexbakervr I'm closing this issue by assuming this feature has been supported as explained [here](https://github.com/hashicorp/terraform-provider-azurerm/issues/15094#issuecomment-1026558617). Feel free to suggest to reopen if you believe there are still gaps.
